plugins/woocommerce-blocks/assets/js/blocks/add-to-cart-with-options/edit/edit-template-part.tsx

```diff
@@ -0,0 +1,129 @@
+/**
+ * External dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { Spinner } from '@wordpress/components';
+import { useEntityBlockEditor, store as coreStore } from '@wordpress/core-data';
+import {
+	InnerBlocks,
+	useInnerBlocksProps,
+	useBlockProps,
+} from '@wordpress/block-editor';
+
+export type FeaturesKeys = 'isBlockifiedAddToCart';
+
+export type FeaturesProps = {
+	[ key in FeaturesKeys ]?: boolean;
+};
+
+export type UpdateFeaturesType = ( key: FeaturesKeys, value: boolean ) => void;
+
+const TemplatePartInnerBlocks = ( {
+	blockProps,
+	templatePartId,
+}: {
+	blockProps: Record< string, unknown >;
+	templatePartId: string | undefined;
+} ) => {
+	const [ blocks, onInput, onChange ] = useEntityBlockEditor(
+		'postType',
+		'wp_template_part',
+		{ id: templatePartId }
+	);
+
+	const { isLoading } = useSelect(
+		( select ) => {
+			const { hasFinishedResolution } = select( coreStore );
+
+			const hasResolvedEntity = templatePartId
+				? hasFinishedResolution( 'getEditedEntityRecord', [
+						'postType',
+						'wp_template_part',
+						templatePartId,
+				  ] )
+				: false;
+
+			return {
+				isLoading: ! hasResolvedEntity,
+			};
+		},
+		[ templatePartId ]
+	);
+
+	const innerBlocksProps = useInnerBlocksProps( blockProps, {
+		value: blocks,
+		onInput,
+		onChange,
+		renderAppender: () =>
+			! isLoading && blocks.length === 0
+				? InnerBlocks.ButtonBlockAppender
+				: null,
+	} );
+
+	if ( isLoading ) {
+		return (
+			<div { ...blockProps }>
+				<Spinner />
+			</div>
+		);
+	}
+
+	return <div { ...innerBlocksProps } />;
+};
+
+export const AddToCartWithOptionsEditTemplatePart = ( {
+	productType,
+}: {
+	productType: string;
+} ) => {
+	const currentTheme = useSelect(
+		( select ) => select( coreStore ).getCurrentTheme()?.stylesheet,
+		[]
+	);
+	const templatePartSlug = `${ productType }-product-add-to-cart-with-options`;
```

Comment:
If these variables `templatePartSlug`, `themeTemplatePartId` and `wooCommerceTemplatePartId` are not going to be used elsewhere they could be moved within the `useSelect` callback function and use `productType` and `currentTheme` as deps, so while we scope them to the callback block we are also more clear about what are the real deps of the `useSelect` hook.

Reply:
Yeah, good point. I moved these variables inside the `useSelect` hook in 02c0b65ea30ec2d9a395af481c2360539ed8b123.